src/middleware/packages/ldp/mixins/orphan-files-deletion.js

```diff
@@ -0,0 +1,62 @@
+const { CronJob } = require('cron');
+const urlJoin = require('url-join');
+
+module.exports = {
+  settings: {
+    orphanFilesDeletion: {
+      baseUrl: undefined,
+      cronJob: {
+        time: '0 0 4 * * *', // Every night at 4am
+        timeZone: 'Europe/Paris'
+      }
+    }
+  },
+  dependencies: ['triplestore', 'ldp.resource'],
+  actions: {
+    async checkOrphanFiles(ctx) {
+      try {
+        this.logger.info('OrphanFilesDeletion - Check...');
+
+        const containerUri = urlJoin(this.settings.orphanFilesDeletion.baseUrl, this.settings.path);
+        const orphanFiles = await ctx.call('triplestore.query', {
+          query: `
+            SELECT ?file
+            WHERE {
+              <${containerUri}> <http://www.w3.org/ns/ldp#contains> ?file .
+              FILTER NOT EXISTS {
+                ?s ?p ?file .
+                FILTER(?s != <${containerUri}>)
+              }
+            }
+          `,
+          webId: 'system'
+        });
+
+        this.logger.info(`OrphanFilesDeletion - Found ${orphanFiles.length} orphan files`);
+
+        for (let orphanFile of orphanFiles) {
```

Comment:
IMO `orphanFiles` is not so well named as it seems to imply it is the list of files.
What we do in other places is that we name it `results` or `nodes`, and then do something like this:
```suggestion
        for (const { file } of results) {
```